### PR TITLE
BC-7320 - Implementation Socket Error Handling for Students

### PR DESCRIPTION
**Pull request by muratmerdoglu-dp**
# Short Description

## Links to Ticket and related Pull-Requests

- https://ticketsystem.dbildungscloud.de/browse/BC-7320

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/modules/feature/board/board/Board.vue [hunk 1]
```diff
@@ -292,14 +292,26 @@ const setAlert = useDebounceFn(() => {
...
 }, 150);
 
-watch(() => board.value?.isVisible, setAlert, { immediate: true });
+watch(
+	() => board.value?.isVisible,
+	() => {
+		setAlert();
+
+		if (!(isBoardVisible.value || isTeacher)) {
+			router.replace({ path: "/rooms/" + roomId.value });
```
Comment (NFriedo) [inline src/modules/feature/board/board/Board.vue:308]: Same as in Board.store.ts (see other comment)
Maybe we should use:
```ts
router.replace({ name: 'rooms', params: { roomId.value } })
```

File: src/modules/feature/board/board/Board.vue [hunk 2]
```diff
@@ -292,14 +292,26 @@ const setAlert = useDebounceFn(() => {
...
-	if (!board.value.isVisible) {
+	if (!isBoardVisible.value) {
 		showCustomNotifier(t("components.board.alert.info.draft"), "info");
 	} else {
 		showCustomNotifier(t("components.board.alert.info.teacher"), "info");
 	}
 }, 150);
 
-watch(() => board.value?.isVisible, setAlert, { immediate: true });
+watch(
```
Comment (NFriedo) [inline src/modules/feature/board/board/Board.vue:302]: Before there was a `{immediate: true}` .
Just curious why do we don't need this anymore?

File: src/modules/data/board/Board.store.ts [hunk 3]
```diff
@@ -43,6 +49,7 @@ export const useBoardStore = defineStore("boardStore", () => {
 	const socketOrRest = isSocketEnabled ? useBoardSocketApi() : restApi;
 
 	const { setEditModeId } = useSharedEditMode();
+	const router = useRouter();
```
Comment (NFriedo) [inline src/modules/data/board/Board.store.ts:52]: Probably this line causes vue warnings in boardRestApi and boardSocketApi tests, see comments below.

File: src/modules/feature/board/board/Board.vue (hunk 2 above)
Comment (muratmerdoglu-dp) [reply src/modules/feature/board/board/Board.vue:302]: There was an error when the `{ immediate: true}` option was set. I think the values and methods are not ready when the watcher is invoked on the initialization. I also think that there is no advantage to using the immediate option before

File: src/modules/feature/board/board/Board.vue [hunk 4]
```diff
@@ -292,14 +292,26 @@ const setAlert = useDebounceFn(() => {
...
+	() => board.value?.isVisible,
```
Comment (NFriedo) [inline src/modules/feature/board/board/Board.vue:303]: Can we use the new computed `isBoardVisible` here?

**Review by NFriedo — COMMENTED** [review]
looks good, just need to clarify with Anna how the birds should look :)

File: src/modules/data/board/Board.store.ts [hunk 5]
```diff
@@ -29,20 +31,28 @@ import { useBoardSocketApi } from "./boardActions/boardSocketApi.composable";
...
+import { HttpStatusCode } from "@/store/types/http-status-code.enum";
+import { useRouter } from "vue-router";
+import { useI18n } from "vue-i18n";
 
 export const useBoardStore = defineStore("boardStore", () => {
 	const cardStore = useCardStore();
 	const board = ref<Board | undefined>(undefined);
 	const isLoading = ref<boolean>(false);
 	const { setFocus, forceFocus } = useBoardFocusHandler();
+	const referenceRoomId = ref<string | undefined>(undefined);
```
Comment (hoeppner-dataport) [inline src/modules/data/board/Board.store.ts:44]: It's unusual to carry the type (="reference") in the name.
Suggestion: rename to "roomId"

File: src/modules/data/board/boardActions/boardRestApi.composable.unit.ts [hunk 6]
```diff
@@ -32,13 +41,28 @@ const mockedSharedEditMode = jest.mocked(useSharedEditMode);
 jest.mock("../socket/socket");
 const mockedUseSocketConnection = jest.mocked(useSocketConnection);
 
+jest.mock("vue-router");
+const useRouterMock = <jest.Mock>useRouter;
+
+jest.mock("vue-i18n", () => {
```
Comment (hoeppner-dataport) [inline src/modules/data/board/boardActions/boardRestApi.composable.unit.ts:47]: I think we do not mock vue-i18n this way anymore. Have a look at i18nMock (e.g. in src/modules/data/provisioning-options/ProvisioningOptionsState.composable.unit.ts)

File: src/modules/feature/board/board/Board.unit.ts [hunk 7]
```diff
@@ -810,6 +838,92 @@ describe("Board", () => {
 
 				expect(boardStore.updateBoardVisibilityRequest).toHaveBeenCalled();
 			});
+
+			describe("@createApplicationError", () => {
+				describe("when board is draft mode", () => {
```
Comment (hoeppner-dataport) [inline src/modules/feature/board/board/Board.unit.ts:843]:
```suggestion
				describe("when board is in draft mode", () => {
```

**Review by hoeppner-dataport — APPROVED** [review]
Found only some smaller things...

**Review by hoeppner-dataport — APPROVED** [review]

**Pull request merged**